### PR TITLE
React capstone project - airPurity app

### PR DESCRIPTION
**Pull request by rudicarrilloypr**
### General requirements

- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### HTML/CSS & JavaScript requirements

- Follow our list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- Follow our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### Project requirements

**API**
First you need to choose an API to base the development of the webapp on. The API should allow you to get numeric values based on a parameter. We recommend that you choose an API that is open or needs an API key. If you choose an API that require authentication, **you should implement it on your own**. 

  Some example APIs are:
  - [Financial modeling prep API](https://financialmodelingprep.com/developer/docs/): stock data.
  - [Air pollution API](https://openweathermap.org/api/air-pollution): air quality data.

  You can find many APIs in [this GitHub repo](https://github.com/public-apis/public-apis). Some of the APIs requires a token for authentication and some others are just open.

**Design**
- You should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
  - Colors (select one main color for your website).
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

**Interactions**
- Home page
  - When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
  - Along with the category name, you will display some numeric values per category that come from the API.
  - When the user clicks (or taps) on a category item, the application navigates to the details page.

- Details page
  - In the details page, the webapp retrieves data from the API to show detailed data in the category.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

**Testing requirements**
- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
  - You may need to mock the access to the API, so that your tests don't send actual requests.
  - You may need to mock the connection to the Redux Store.

**Technical requirements**

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- You should filter the data that you retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project should be deployed and accessible online using [Render](https://render.com), [Netlify](https://www.netlify.com/) or [GitHub Pages](https://pages.github.com/).
  - Add a link to the online version of your app to your README file.

### Project documentation

Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a **maximum length of 5 minutes**. The content of the video should include:

- a description of the project.
- a demo of the project features.
- you should also highlight some interesting piece of code or something you built that you are very proud of.

For recording the video you can use tools like [Loom](https://www.loom.com/) that let you share a private link to the recording, and configure a shot that shows your computer screen and your face at the same time in a small picture.

### CODE REVIEW

File: src/components/DetailsPage.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
+/* eslint-disable react/prop-types */
```
Comment (Reem-lab) [inline src/components/DetailsPage.js:1]: - [ ] kindly ensure that no linter is disabled, linters are tools that analyze your code to identify and flag potential issues, errors, or stylistic inconsistencies. They are widely used in software development to ensure code quality and maintainability. so kindly try to solve the issue not disable it 💯💪🏻⭐ I am sure you will find a solution for it 👏🏻💪🏻💯

File: src/components/Filter.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
+/* eslint-disable max-len */
+/* eslint-disable no-alert */
```
Comment (Reem-lab) [inline src/components/Filter.js:2]: - [ ]  kindly ensure that no linter is disabled, linters are tools that analyze your code to identify and flag potential issues, errors, or stylistic inconsistencies. They are widely used in software development to ensure code quality and maintainability. so kindly try to solve the issue not disable it 💯💪🏻⭐ I am sure you will find a solution for it 👏🏻💪🏻💯

File: src/components/HomePage.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+/* eslint-disable react/prop-types */
```
Comment (Reem-lab) [inline src/components/HomePage.js:1]: - [ ]  kindly ensure that no linter is disabled, linters are tools that analyze your code to identify and flag potential issues, errors, or stylistic inconsistencies. They are widely used in software development to ensure code quality and maintainability. so kindly try to solve the issue not disable it 💯💪🏻⭐ I am sure you will find a solution for it 👏🏻💪🏻💯

File: src/components/HomePage.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <Filter fetchCoordinates={fetchCoordinates} selectState={selectState} />
+      <ul>
+        {Array.isArray(items.list) ? items.list.map((item, index) => (
+          // eslint-disable-next-line react/no-array-index-key
```
Comment (Reem-lab) [inline src/components/HomePage.js:28]: - [ ]  kindly ensure that no linter is disabled, linters are tools that analyze your code to identify and flag potential issues, errors, or stylistic inconsistencies. They are widely used in software development to ensure code quality and maintainability. so kindly try to solve the issue not disable it 💯💪🏻⭐ I am sure you will find a solution for it 👏🏻💪🏻💯

File: src/styles/filter.css [hunk 5]
```diff
@@ -0,0 +1,272 @@
+/* stylelint-disable no-missing-end-of-source-newline */
```
Comment (Reem-lab) [inline src/styles/filter.css:1]: - [ ] kindly ensure that no linter is disabled, linters are tools that analyze your code to identify and flag potential issues, errors, or stylistic inconsistencies. They are widely used in software development to ensure code quality and maintainability. so kindly try to solve the issue not disable it 💯💪🏻⭐ I am sure you will find a solution for it 👏🏻💪🏻💯

File: src/tests/DetailsPage.test.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
+/* eslint-disable max-len */
```
Comment (Reem-lab) [inline src/tests/DetailsPage.test.js:1]: - [ ] kindly ensure that no linter is disabled, linters are tools that analyze your code to identify and flag potential issues, errors, or stylistic inconsistencies. They are widely used in software development to ensure code quality and maintainability. so kindly try to solve the issue not disable it 💯💪🏻⭐ I am sure you will find a solution for it 👏🏻💪🏻💯

File: src/tests/actions.test.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable max-len */
```
Comment (Reem-lab) [inline src/tests/actions.test.js:1]: - [ ] kindly ensure that no linter is disabled, linters are tools that analyze your code to identify and flag potential issues, errors, or stylistic inconsistencies. They are widely used in software development to ensure code quality and maintainability. so kindly try to solve the issue not disable it 💯💪🏻⭐ I am sure you will find a solution for it 👏🏻💪🏻💯

File: src/reducers/items.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+      // Aquí se actualiza allItems
+    case FETCH_ITEMS_FAILURE:
+      return { ...state, loading: false, error: action.payload };
+    case APPLY_FILTER:
+      return {
+        ...state,
+        filter: action.payload,
+        // Aquí utilizamos allItems en lugar de initialState.items
+        // eslint-disable-next-line max-len
```
Comment (Reem-lab) [inline src/reducers/items.js:48]: - [ ]  kindly ensure that no linter is disabled, linters are tools that analyze your code to identify and flag potential issues, errors, or stylistic inconsistencies. They are widely used in software development to ensure code quality and maintainability. so kindly try to solve the issue not disable it 💯💪🏻⭐ I am sure you will find a solution for it 👏🏻💪🏻💯

File: src/reducers/items.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    case FETCH_ITEMS_SUCCESS:
+      return {
+        ...state,
+        loading: false,
+        items: Array.isArray(action.payload.list) ? action.payload.list : [],
+        allItems: Array.isArray(action.payload.list) ? action.payload.list : [],
+      };
+
+      // Aquí se actualiza allItems
+    case FETCH_ITEMS_FAILURE:
```
Comment (Reem-lab) [inline src/reducers/items.js:41]: - [ OPTIONAL ] I highly recommend adding your reducer folder and store.js file and action file in one directory called redux to organize your files structure 🤩🔥💪🏻

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @rudicarrilloypr, 👋🏻👋🏻

Good job so far! 💯⭐🔥🚀🤩
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Great working on the styles ✨🎉
- Followed Gitflow 🙏🏻
- Professional README ✨
- Descriptive PR 💯💯
- Good working on the functionality for getting the data from the API 🔥
- Amazing for displaying the images for each region by JSON file 🔥
- Deployed well 🤩🚀
- Great working on the tests, All passed ✅✅
- Great working on the video presentation and mentioning the part you are proud ⭐

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Owusu-Desmond — APPROVED** [review]
Hi @rudicarrilloypr , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/reducers/items.js (hunk 9 above)
Comment (rudicarrilloypr) [reply src/reducers/items.js:46]: thanks for the support and review! @Reem-lab

**Pull request merged**